front/src/modules/trainschedule/components/TimetableV2/TimetableToolbar.tsx

```diff
@@ -114,6 +116,31 @@ const TimetableToolbar = ({
       });
   };
 
+  const exportSchedules = async (selectedTrainIdsFromClick: number[]) => {
+    const trainScheduleResults = await getV2TrainSchedule({ ids: selectedTrainIdsFromClick })
+      .unwrap()
+      .then((results) =>
+        results.map((result) => {
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          const { id, ...trainSchedule } = result;
+          return trainSchedule;
+        })
+      )
+      .catch((e) => {
+        dispatch(setFailure(castErrorToFailure(e)));
+      });
+
+    if (!trainScheduleResults) return;
+
+    const jsonString = JSON.stringify({ trainScheduleResults });
```

Comment:
It's easy to miss it if you didn't work on this part, but we are already fetching trainSchedules in the `useTrainSchedulesDetails` hook in this file.
To avoid a second call when exporting the results, you can add `trainSchedules` in the return datas of the hook to use it in your export function.

```suggestion
      if (!trainSchedules) return;

    const formattedTrainSchedules = trainSchedules
      .filter(({ id }) => selectedTrainIdsFromClick.includes(id))
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      .map(({ id, ...trainSchedule }) => trainSchedule);

    const jsonString = JSON.stringify({ train_schedules: formattedTrainSchedules });
```

Reply:
Done